packages/ra-ui-materialui/src/input/ResettableTextField.js

```diff
@@ -30,107 +28,110 @@ const styles = createStyles({
 /**
  * An override of the default Material-UI TextField which is resettable
  */
-class ResettableTextField extends Component {
-    static propTypes = {
-        classes: PropTypes.object.isRequired,
-        clearAlwaysVisible: PropTypes.bool,
-        disabled: PropTypes.bool,
-        InputProps: PropTypes.object,
-        onBlur: PropTypes.func,
-        onChange: PropTypes.func.isRequired,
-        onFocus: PropTypes.func,
-        resettable: PropTypes.bool,
-        translate: PropTypes.func.isRequired,
-        value: PropTypes.any.isRequired,
-    };
-
-    state = { showClear: false };
+function ResettableTextField({
+    classes: classesOverride,
+    clearAlwaysVisible,
+    InputProps,
+    value,
+    resettable,
+    disabled,
+    variant = 'filled',
+    margin = 'dense',
+    ...props
+}) {
+    const [showClear, setShowClear] = useState(false);
+    const classes = useStyles({ classes: classesOverride });
+    const translate = useTranslate();
 
-    handleClickClearButton = event => {
-        event.preventDefault();
-        this.props.onChange('');
-    };
+    const { onChange, onFocus, onBlur } = props;
+    const handleClickClearButton = useCallback(
+        event => {
+            event.preventDefault();
+            onChange('');
+        },
+        [onChange]
+    );
 
-    handleMouseDownClearButton = event => {
+    const handleMouseDownClearButton = useCallback(event => {
```

Comment:
If it doesn't use anything from the component itself (such as state or props), you can even declare this function outside of the component and won't need the `useCallback` at all

Reply:
ok, thanks! I've declared it outside the component.